src/components/encoder/Client.ts

```diff
@@ -0,0 +1,80 @@
+import { Struct } from 'google-protobuf/google/protobuf/struct_pb';
+import type { RobotClient } from '../../robot';
+import type { Encoder } from './Encoder';
+import { EncoderServiceClient } from '../../gen/component/encoder/v1/encoder_pb_service';
+import { type Options, PositionType } from '../../types';
+import encoderApi from '../../gen/component/encoder/v1/encoder_pb';
+import { promisify } from '../../utils';
+
+/**
+ * A gRPC-web client for the Encoder component.
+ *
+ * @group Clients
+ */
+export class EncoderClient implements Encoder {
+  private client: EncoderServiceClient;
+  private readonly name: string;
+  private readonly options: Options;
+
+  constructor(client: RobotClient, name: string, options: Options = {}) {
+    this.client = client.createServiceClient(EncoderServiceClient);
+    this.name = name;
+    this.options = options;
+  }
+
+  private get encoderService() {
+    return this.client;
+  }
+
+  async resetPosition(extra = {}) {
+    const encoderService = this.encoderService;
+    const request = new encoderApi.ResetPositionRequest();
+    request.setName(this.name);
+    request.setExtra(Struct.fromJavaScript(extra));
+
+    this.options.requestLogger?.(request);
+
+    await promisify<
+      encoderApi.ResetPositionRequest,
+      encoderApi.ResetPositionResponse
+    >(encoderService.resetPosition.bind(encoderService), request);
+  }
+
+  async getProperties(extra = {}) {
+    const encoderService = this.encoderService;
+    const request = new encoderApi.GetPropertiesRequest();
+    request.setName(this.name);
+    request.setExtra(Struct.fromJavaScript(extra));
+
+    this.options.requestLogger?.(request);
+
+    const response = await promisify<
+      encoderApi.GetPropertiesRequest,
+      encoderApi.GetPropertiesResponse
+    >(encoderService.getProperties.bind(encoderService), request);
+    return {
+      ticksCountSupported: response.getTicksCountSupported(),
+      angleDegreesSupported: response.getAngleDegreesSupported(),
+    };
+  }
+
+  async getPosition(positionType?: PositionType, extra = {}) {
+    const encoderService = this.encoderService;
+    const request = new encoderApi.GetPositionRequest();
+    request.setName(this.name);
+    if (positionType === undefined) {
+      request.setPositionType(PositionType.POSITION_TYPE_UNSPECIFIED);
+    } else {
+      request.setPositionType(positionType);
+    }
```

Comment:
alternatively we can set `POSITION_TYPE_UNSPECIFIED` as the default value:

```suggestion
  async getPosition(positionType: PositionType = PositionType.POSITION_TYPE_UNSPECIFIED, extra = {}) {
    const encoderService = this.encoderService;
    const request = new encoderApi.GetPositionRequest();
    request.setName(this.name);
    request.setPositionType(positionType);
```

Reply:
it's actually equivalent to what you have - it will use the default if you omit the value [e.g. `getPosition()`] OR if pass in `undefined` as the first arg [e.g. `getPosition(undefined)`].